### PR TITLE
No XML because files can contain ampersands and double quotes

### PR DESCRIPTION
**Pull request by pepa65**
Files or directories can have ampersands (common) or double quotes (rarer) in their names, and this breaks XML (Firefox gives an error page). There is no reason to do XML, so went to HTML 4.01 transitional, while retaining the css.

### CODE REVIEW

**Comment by pepa65** [comment]
Also, added a newline before <html> because otherwise it trips up elinks and w3m browsers.

File: utils.c [hunk 1]
```diff
@@ -108,34 +110,31 @@ int list_dir(connection_t *connection_prop, char *html, unsigned int bufsize, bo
...
         } else if (S_ISDIR(f_mode)) { //Directory entry
             //Table row for the dir
             printf_s=snprintf(html+pagesize,maxsize,
-                              "<tr style=\"background-color: #DFDFDF;\"><td>d</td><td><a href=\"%s/\">%s/</a></td><td>-</td><td>%s</td></tr>\n",
-                              namelist[i]->d_name, namelist[i]->d_name,last_modified);
+                              "<tr style=\"background-color: %s;\"><td><b>d</b></td><td><a href=\"%s\"><b>%s/</b></a></td><td></td><td>%s</td></tr>\n",
+                              color, namelist[i]->d_name, namelist[i]->d_name,last_modified);
             maxsize-=printf_s;
             pagesize+=printf_s;
+            color = (color == "white" ? "#E0E0E0" : "white");
```
Comment (ltworf) [inline utils.c]: Eh, but that no longer gives directories a different colour than normal files.

Comment (pepa65) [inline utils.c]: If that is important to you, I would recommend setting the color on the directories like #D0D0D0 as before. But then you no longer have alternating light/dark lines if the directories fall on a light row... Instead, I made the font bold, so they stand out that way.

But, I don't mean to push all these things onto you, that's just personal taste. There are 2 things that I think are wrong now:
1. Using XML headers (That was the first commit, and I did a pull request for that one. I didn't mean for you to pull all the other changes...)
2. The time says GMT, which is only correct if the local time zone is GMT. (I live in Thailand...) I didn't know how to send that commit as a separate issue... Sorry, not so handy with git.

**Comment by pepa65** [comment]
Sorry, I now understand I should have put things in different branches...
If you can give some pointers how I can send you these 2 commits, I'll do it:
cbd3da392f6a77586a0f11e9ce490daa41222af1
c401f036f60d9a40407cc5be2321762a2a01b69e

File: utils.c (hunk 1 above)
Comment (ltworf) [inline utils.c]: I agree with that so I have merged these commits, but haven't merged the commits about the colours, plus all the other ones you made after I had merged them.

**Comment by ltworf** [comment]
Rebase on `origin/stable`, apply those commits.

If you run `git format-patch origin/stable` it will create one file per commit, then you can mail them to me.

If you do this often enough, you can install git mail to automate the mail sending part.

File: utils.c (hunk 1 above)
Comment (pepa65) [inline utils.c]: Excellent! Then I will just proceed to customize the code for my purpose for now, please ignore further changes. If I find anything worthy of inclusion for you, I'll do it on a fresh rebase.

File: options.h.in [hunk 2]
```diff
@@ -74,7 +74,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+#define HTMLHEAD "<!DOCTYPE HTML PUBLIC \"-//W3C//DTD HTML 4.01 Transitional//EN\">\n<html><head><meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" /><meta http-equiv=\"pragma\" content=\"no-cache\" /><title>" NAME "</title><style type=\"text/css\">a, a:active {text-decoration: none; color: blue;}a:visited {color: #48468F;}a:hover, a:focus {text-decoration: underline; color: red;}body {background-color: #F5F5F5;}h2 {margin-bottom: 12px;}table {margin-left: 12px;}th, td { font: 80% sans-serif; text-align: left;}th { font-weight: bold; padding-right: 14px; padding-bottom: 3px;}td {padding-right: 14px;}td.s, th.s {text-align: right;}div.list { background-color: white; border-top: 1px solid #646464; border-bottom: 1px solid #646464; padding-top: 10px; padding-bottom: 14px;}div.foot { font: 80% sans-serif; color: #787878; padding-top: 4px;}</style></head><body><h4>"
```
Comment (ltworf) [inline options.h.in]: weborf uses ETag for generated content, does this play nice with that in the browsers?

Comment (pepa65) [inline options.h.in]: Haven't tried that, I need the directory browser functionality. (That's why I also need the authentication bit..)

**Comment by ltworf** [comment]
Closing this so I stop getting notifications. Send relevant commits over email please.

File: options.h.in (hunk 2 above)
Comment (pepa65) [inline options.h.in]: So I noticed elinks isn't able to handle the ETags well, and w3m not at all, same as links and links2. Lynx can handle them, as can Netrik. Lynx can also handle the embedded authentication.

**Comment by pepa65** [comment]
Not sure what all you're interested in. I have one for showing the directory tree op top of the page instead of the Weborf text. And I have integrated the embedded authentication through a username/password command line option so you can choose which one you want to use without recompiling.

**Comment by ltworf** [comment]
Improvements: yes
Change "weborf" to "Thai server": no :D

**Comment by pepa65** [comment]
That was for our office, but it's not very useful for others. :-)
I'm going to do commandline switches for that I think, at least for NAME.

**Pull request closed without merging**